src/library/getSampleEventCounts.ts

```
@@ -0,0 +1,23 @@
+/**
+ * Counts the number of sample events in the records array grouped by protocol
+ *
+ * @param {Array} records - An array of sample event records to filter
+ * @returns {Object} An object mapping each protocol to its count
+ */
+type SampleEventRecord = {
```

ESLint

[error] Use an `interface` instead of a `type`.

(@typescript-eslint/consistent-type-definitions)